### ISSUE TITLE
"java.lang.IllegalStateException: Absent persistent model" when exit Eclipse Oxygen M5

### ISSUE THREAD

**Issue by missedone** [eclipse/buildship#411]
I got the following error when exit Eclipse Oxygen JEE bundle M5 20170202-1353
(Note, not happened every time)

Buildship version:   Buildship: Eclipse Plug-ins for Gradle	2.0.1.v20170220-1739-m	org.eclipse.buildship.feature.group	Eclipse Buildship

```
java.lang.IllegalStateException: Absent persistent model
  at org.eclipse.buildship.core.preferences.internal.AbsentPersistentModel.getBuildDir(AbsentPersistentModel.java:47)
  at org.eclipse.buildship.core.preferences.internal.PersistentModelConverter.toProperties(PersistentModelConverter.java:46)
  at org.eclipse.buildship.core.preferences.internal.DefaultModelPersistence.persistPrefsChecked(DefaultModelPersistence.java:156)
  at org.eclipse.buildship.core.preferences.internal.DefaultModelPersistence.persistPrefs(DefaultModelPersistence.java:142)
  at org.eclipse.buildship.core.preferences.internal.DefaultModelPersistence.persistAllProjectPrefs(DefaultModelPersistence.java:136)
  at org.eclipse.buildship.core.preferences.internal.DefaultModelPersistence.onEvent(DefaultModelPersistence.java:92)
  at org.eclipse.buildship.core.event.internal.DefaultListenerRegistry.dispatch(DefaultListenerRegistry.java:53)
  at org.eclipse.buildship.ui.workspace.ShutdownListener.preShutdown(ShutdownListener.java:26)
  at org.eclipse.ui.internal.Workbench$3.run(Workbench.java:968)
  at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
  at org.eclipse.ui.internal.JFaceUtil.lambda$0(JFaceUtil.java:44)
  at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
  at org.eclipse.ui.internal.Workbench.firePreShutdown(Workbench.java:965)
  at org.eclipse.ui.internal.Workbench.busyClose(Workbench.java:1102)
  at org.eclipse.ui.internal.Workbench.lambda$4(Workbench.java:1428)
  at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
  at org.eclipse.ui.internal.Workbench.close(Workbench.java:1428)
  at org.eclipse.ui.internal.Workbench.close(Workbench.java:1401)
  at org.eclipse.ui.internal.WorkbenchWindow.busyClose(WorkbenchWindow.java:1594)
  at org.eclipse.ui.internal.WorkbenchWindow.access$16(WorkbenchWindow.java:1540)
  at org.eclipse.ui.internal.WorkbenchWindow$11.run(WorkbenchWindow.java:1626)
  at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
  at org.eclipse.ui.internal.WorkbenchWindow.close(WorkbenchWindow.java:1623)
  at org.eclipse.ui.internal.WorkbenchWindow.close(WorkbenchWindow.java:1637)
  at org.eclipse.ui.internal.WorkbenchWindow$6.close(WorkbenchWindow.java:524)
  at org.eclipse.e4.ui.workbench.renderers.swt.WBWRenderer$3.shellClosed(WBWRenderer.java:617)
  at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:99)
  at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
  at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4257)
  at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1501)
  at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1524)
  at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1509)
  at org.eclipse.swt.widgets.Shell.closeWidget(Shell.java:621)
  at org.eclipse.swt.widgets.Shell.windowShouldClose(Shell.java:2433)
  at org.eclipse.swt.widgets.Display.windowProc(Display.java:5795)
  at org.eclipse.swt.internal.cocoa.OS.objc_msgSendSuper(Native Method)
  at org.eclipse.swt.widgets.Widget.callSuper(Widget.java:227)
  at org.eclipse.swt.widgets.Widget.windowSendEvent(Widget.java:2140)
  at org.eclipse.swt.widgets.Shell.windowSendEvent(Shell.java:2402)
  at org.eclipse.swt.widgets.Display.windowProc(Display.java:5861)
  at org.eclipse.swt.internal.cocoa.OS.objc_msgSendSuper(Native Method)
  at org.eclipse.swt.widgets.Display.applicationSendEvent(Display.java:5290)
  at org.eclipse.swt.widgets.Display.applicationProc(Display.java:5439)
  at org.eclipse.swt.internal.cocoa.OS.objc_msgSend(Native Method)
  at org.eclipse.swt.internal.cocoa.NSApplication.sendEvent(NSApplication.java:128)
  at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3695)
  at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1133)
  at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
  at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
  at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
  at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:684)
  at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
  at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:598)
  at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
  at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
  at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
  at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
  at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
  at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
  at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
  at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
  at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
```

**Comment by donat** [comment]
This has been fixed in the 2.0.1 release: https://github.com/eclipse/buildship/issues/404
Unfortunately, Oxygen M5 contains a snapshot version which not yet include the fix. I'll update the contribution. In the meanwhile you can [try installing](https://github.com/eclipse/buildship/blob/master/docs/user/Installation.md) the latest snapshot or release version of Buildship.

**Comment by donat** [comment]
2.0.1 Contributed to Eclipse Oxygen: http://git.eclipse.org/c/simrel/org.eclipse.simrel.build.git/commit/?id=e4c760d6159936ef261c8a1c768449fd05bdef2c
